### PR TITLE
[Filebeat] Improve ECS categorization field mappings for zeek module

### LINKED ISSUE #16029: [Filebeat] Update zeek module to ECS 1.4
### Filesets

  - [ ] kerberos
  - [ ] intel
  - [ ] ftp
  - [ ] pe
  - [ ] ssl
  - [ ] syslog
  - [ ] snmp
  - [ ] connection
  - [ ] smb_mapping
  - [ ] ntlm
  - [ ] dce_rpc
  - [ ] irc
  - [ ] rdp
  - [ ] radius
  - [ ] rfb
  - [ ] notice
  - [ ] tunnel
  - [ ] x509
  - [ ] smb_cmd
  - [ ] ssh
  - [ ] smb_files
  - [ ] http
  - [ ] smtp
  - [ ] mysql
  - [ ] dhcp
  - [ ] files
  - [ ] capture_loss
  - [ ] sip
  - [ ] socks
  - [ ] ocsp
  - [ ] traceroute
  - [ ] dns
  - [ ] modbus
  - [ ] dnp3
  - [ ] weird
  - [ ] dpd
  - [ ] stats

Look at:
- [ ] dns, event.outcome, dns.question.top_level_domain
- [ ] mysql, event.outcome
- [ ] ssh, event.outcome

### PR DESCRIPTION
**Pull request by leehinman**
## What does this PR do?
Updates the filesets of zeek module to ECS 1.5

## Why is it important?
ECS categorization fields improve experience in SIEM module and across inputs from different sources.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
```
TESTING_FILEBEAT_MODULES=zeek mage -v pythonIntegTest
```

## Related issues
Closes #16029

## Changes to each Fileset

- capture_loss
  + convert pipeline to yaml
  + event.kind
  + event.type
- connection
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
- dce_rpc
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.action
  + related.ip
  + source.geo
  + destination.geo
- dhcp
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
- dnp3
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.action
  + related.ip
  + source.geo
  + destination.geo
  + source.as
  + destiantion.as
- dns
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + dns.question.top_level_domain
  + related.ip
- dpd
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + source.geo
  + destiantion.geo
  + source.as
  + destiantion.as
  + related.ip
- files
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + file.mime_type
  + file.name
  + file.hash.md5
  + file.hash.sha1
  + file.hash.sha256
  + client.ip
  + server.ip
  + related.ip
  + related.hash
- ftp
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.action
  + user.name
  + file.mime_type
  + file.size
  + related.ip
  + related.user
  + source.geo
  + destination.geo
- http
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + event.action
  + url.port type to number
  + http.request.method lowercase
  + related.ip
  + related.users
- intel
  + event.kind
  + event.type
  + related.ip
- irc
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.action
  + user.name
  + file.name
  + file.size
  + file.mime_type
  + related.ip
  + related.user
  + source.geo
  + destination.geo
- kerberos
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + event.action
  + user.name
  + user.domain
  + source.geo
  + destination.geo
  + related.ip
  + related.user
- modbus
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + event.action
  + related.ip
  + source.geo
  + destination.geo
- mysql
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + source.geo
  + destiantion.geo
  + related.ip
- notice
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
  + file.size
  + file.mime_type
  + rule.description
  + rule.name
- ntlm
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + user.name
  + user.domain
  + source.geo
  + destination.geo
  + related.ip
  + related.user
- ocsp
  + convert pipeline to yaml
  + event.kind
  + related.hash
- pe
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
- radius
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + user.name
  + source.geo
  + destination.geo
  + related.ip
  + related.user
- rdp
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
  + source.geo
  + destination.geo
- rfb
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
  + source.geo
  + destination.geo
- sip
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + event.action
  + related.ip
  + source.geo
  + destination.geo
  + url.full
- smb_cmd
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + event.action
  + related.ip
  + related.user
  + source.geo
  + destination.geo
  + user.name
- smb_files
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.action
  + related.ip
  + related.user
  + source.geo
  + destination.geo
  + user.name
  + file.accessed
  + file.ctime
  + file.created
  + file.mtime
  + file.path
  + file.name
  + file.size
- smb_mapping
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
  + source.geo
  + destination.geo
- smtp
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
  + source.geo
  + destination.geo
- snmp
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
  + source.geo
  + destination.geo
- socks
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + related.ip
  + source.geo
  + destination.geo
  + user.name
  + related.user
- ssh
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + event.outcome
  + related.ip
  + source.geo
  + destination.geo
- ssl
  + event.kind
  + event.category
  + event.type
  + related.ip
- stats
  + convert pipeline to yaml
  + event.kind
- syslog
  + convert pipeline to yaml
  + event.kind
  + related.ip
  + source.geo
  + destination.geo
  + log.syslog.facility.name
  + log.syslog.severity.name
- traceroute
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
  + source.geo
  + destination.geo
- tunnel
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
  + source.geo
  + destination.geo
- weird
  + convert pipeline to yaml
  + event.kind
  + event.category
  + event.type
  + related.ip
  + source.geo
  + destination.geo
  + rule.name
- x509
  + event.kind
  + event.type

### CODE REVIEW

File: x-pack/filebeat/module/zeek/dns/config/dns.yml [hunk 1]
```diff
@@ -105,12 +110,54 @@ processors:
             evt.Put("event.duration", rttSec * 1000000000);
         }
 
+        function addTopLevelDomain(evt) {
+            var rd = evt.Get("dns.question.registered_domain");
+            if (rd == null) {
```
Comment (andrewkroh) [inline x-pack/filebeat/module/zeek/dns/config/dns.yml:115]:
```suggestion
            if (!rd) {
```

This covers both null and undefined.

File: x-pack/filebeat/module/zeek/files/ingest/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- append:
+    field: server.ip
+    value: "{{zeek.files.tx_host}}"
+    if: "ctx?.zeek?.files?.tx_host != null"    
+- append:
+    field: related.ip
+    value: "{{zeek.files.tx_host}}"
+    if: "ctx?.zeek?.files?.tx_host != null"
+- append:
+    field: client.ip
```
Comment (andrewkroh) [inline x-pack/filebeat/module/zeek/files/ingest/pipeline.yml:38]: Should this be a `set`? I thought client.ip was not an array. Same for `server.ip`.

File: x-pack/filebeat/module/zeek/kerberos/config/kerberos.yml [hunk 3]
```diff
@@ -72,7 +72,33 @@ processors:
...
+  - dissect:
+      when:
+        contains:
+          zeek.kerberos.client: "/"
+      tokenizer: "%{user.name}/%{user.domain}"
+      field: zeek.kerberos.client
+      target_prefix: ""
 {{ if .community_id }}
   - community_id:
       fields:
```
Comment (andrewkroh) [inline x-pack/filebeat/module/zeek/kerberos/config/kerberos.yml:104]: You could drop the `fields` here too now.

File: x-pack/filebeat/module/zeek/ntlm/config/ntlm.yml [hunk 4]
```diff
@@ -48,7 +48,39 @@ processors:
...
+      equals:
+        zeek.ntlm.success: false
+    then:
+      - add_fields:
+          target: event
+          fields:
+            outcome: failure
 {{ if .community_id }}
   - community_id:
       fields:
```
Comment (andrewkroh) [inline x-pack/filebeat/module/zeek/ntlm/config/ntlm.yml:86]: `fields` can be removed. The defaults will work I think.

File: x-pack/filebeat/module/zeek/files/ingest/pipeline.yml (hunk 2 above)
Comment (leehinman) [reply x-pack/filebeat/module/zeek/files/ingest/pipeline.yml:38]: yeah, I was waffling on this.  current ECS doc for client.ip is:
```
IP address of the client.
Can be one or multiple IPv4 or IPv6 addresses.
type: ip
```
@webmat thoughts?

Comment (webmat) [reply x-pack/filebeat/module/zeek/files/ingest/pipeline.yml:38]: Oh no, Lee you found another hidden landmine in ECS!

I agree with Andrew, I think this should be a single value. This was discussed here recently https://github.com/elastic/ecs/issues/786 as well.

Did you have a concrete use for this being an array?

Comment (leehinman) [reply x-pack/filebeat/module/zeek/files/ingest/pipeline.yml:38]: zeek files has transmit & receive hosts that can be arrays.
[zeek files doc](https://docs.zeek.org/en/current/scripts/base/frameworks/files/main.zeek.html#type-Files::Info)

currently we set the first element of the tx_host to server.ip & first element of rx_host to client.ip.

I'm good with keeping it one value.  I'll just make sure all the tx_hosts & rx_hosts get set in related.ip

**Pull request merged**